components/AdventurerStatus.ts

```diff
@@ -215,4 +222,109 @@ export class AdventurerStatus extends HTMLElement {
   }
 }
 
+import { getFlowState } from '../game/utils';
+
 customElements.define('adventurer-status', AdventurerStatus);
+
+export class FlowChart extends HTMLElement {
+  private _skill: number = 50;
+  private _challenge: number = 50;
+  private _canvas: HTMLCanvasElement | null = null;
+  private _ctx: CanvasRenderingContext2D | null = null;
+
+  static get observedAttributes() {
+    return ['skill', 'challenge'];
+  }
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    this.innerHTML = `
+      <div class="relative" style="width: 100px; height: 100px;">
+        <canvas width="100" height="100"></canvas>
+      </div>
+    `;
+    this._canvas = this.querySelector('canvas');
+    this._ctx = this._canvas!.getContext('2d');
+    this.render();
+  }
+
+  attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+    if (oldValue === newValue) return;
+    if (name === 'skill') {
+      this._skill = parseFloat(newValue);
+    }
+    if (name === 'challenge') {
+      this._challenge = parseFloat(newValue);
+    }
+    this.render();
+  }
+
+  render() {
+    if (!this._ctx || !this._canvas) return;
+
+    const ctx = this._ctx;
+    const canvas = this._canvas;
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
+
+    for (let x = 0; x < 100; x++) {
+      for (let y = 0; y < 100; y++) {
+        const flowState = getFlowState(x, 100 - y);
+        ctx.fillStyle = this.getFlowStateCanvasColor(flowState);
+        ctx.fillRect(x, y, 1, 1);
+      }
+    }
+
+    const dotX = Math.max(0, Math.min(100, this._skill));
+    const dotY = 100 - Math.max(0, Math.min(100, this._challenge));
+    ctx.beginPath();
+    ctx.arc(dotX, dotY, 3, 0, 2 * Math.PI, false);
+    ctx.fillStyle = 'white';
+    ctx.fill();
+    ctx.lineWidth = 1;
+    ctx.strokeStyle = 'black';
+    ctx.stroke();
```

Comment:
move the plotting of the dot to the end of the render function so it appears above everything else